Launching gz_sim.launch.py fails

I ran the below command to launch [gz_sim.launch.py](https://github.com/gazebosim/ros_gz/blob/ros2/ros_gz_sim/launch/gz_sim.launch.py.in) as stated in the [Gazebo docs](https://gazebosim.org/docs/harmonic/ros2_launch_gazebo/#using-the-launch-files-included-in).

```bash
ros2 launch ros_gz_sim gz_sim.launch.py gz_args:=empty.sdf
```

I receive the following error, which prevents Gazebo from launching:

```bash
[INFO] [gazebo-1]: process started with pid [54955]
[gazebo-1] /nix/store/pqrs2y7qkcm12irxmnvyapw42c4jskrb-ros-env/bin/gz: /nix/store/pqrs2y7qkcm12irxmnvyapw42c4jskrb-ros-env/bin/gz:1: Invalid char `\x7F' in expression (SyntaxError)
[gazebo-1] /nix/store/pqrs2y7qkcm12irxmnvyapw42c4jskrb-ros-env/bin/gz:1: Invalid char `\x02' in expression
[gazebo-1] /nix/store/pqrs2y7qkcm12irxmnvyapw42c4jskrb-ros-env/bin/gz:1: Invalid char `\x01' in expression
[gazebo-1] /nix/store/pqrs2y7qkcm12irxmnvyapw42c4jskrb-ros-env/bin/gz:1: Invalid char `\x01' in expression
[ERROR] [gazebo-1]: process has died [pid 54955, exit code 1, cmd 'ruby /nix/store/pqrs2y7qkcm12irxmnvyapw42c4jskrb-ros-env/bin/gz sim empty.sdf --force-version 8'].
```

This error does not occur on Ubuntu.

I tried to run the command, the launch file is running without using Ruby, it works:

```
/nix/store/pqrs2y7qkcm12irxmnvyapw42c4jskrb-ros-env/bin/gz sim empty.sdf --force-version 8
```

--- a/distros/ros2-overlay.nix
+++ b/distros/ros2-overlay.nix
@@ -273,8 +273,8 @@ rosSelf: rosSuper: with rosSelf.lib; {
     # in our case it is an regular executable because it is wrapped.
     postPatch = postPatch + ''
       substituteInPlace launch/gz_sim.launch.py.in \
-        --replace-warn 'ruby $(which gz) sim' 'gz sim' \
-        --replace-warn 'ruby $(which ign) gazebo' 'ign gazebo'
+        --replace-fail "'ruby ' + get_executable_path('gz') + ' sim'" "'gz sim'" \
+        --replace-fail "'ruby ' + get_executable_path('ign') + ' gazebo'" "'ign gazebo'"
     '';
   });
 

